subiquity/server/controllers/ad.py

```diff
@@ -91,9 +91,49 @@ def has_support(self) -> bool:
 
 class ADController(SubiquityController):
     """ Implements the server part of the Active Directory feature. """
-    model_name = "ad"
     endpoint = API.active_directory
     # No auto install key and schema for now due password handling uncertainty.
+    autoinstall_key = model_name = "ad"
+    autoinstall_schema = {
+        'type': 'object',
+        'properties': {
+            'admin_name': {
+                'type': 'string',
+            },
+            'domain_name': {
+                'type': 'string',
+            },
+        },
+        'additionalProperties': False,
+    }
+    autoinstall_default = {"admin_name": '', 'domain_name': ''}
+
+    def serialize(self):
+        info = self.model.conn_info
+        if info is None:
+            return None
+
+        return {'admin_name': info.admin_name, 'domain_name': info.domain_name}
+
+    def deserialize(self, state):
+        if state is None:
+            return
+        if 'admin_name' in state and 'domain_name' in state:
```

Comment:
serialize/deserialize is part of how Subiquity self-update works, or any other service restart.   Having a partial serialize is a bit odd.  While we have generally added serialize/deserialize to new controllers it isn't actually required, and other common ones like identity and filesystem that handle sensitive information don't provide it.

Probably better to just not have serialize/deserialize methods, and if a restart happens subiquity will just show the screen again.